Objects/genericaliasobject.c

```diff
@@ -626,6 +639,50 @@ static PyNumberMethods ga_as_number = {
         .nb_or = _Py_union_type_or, // Add __or__ function
 };
 
+static PyObject *
+ga_iternext(gaiterobject *gi) {
+    if (gi->obj == NULL) {
+        return NULL;
+    }
+    gaobject *alias = (gaobject *) gi->obj;
+    PyObject *starred_tuple = Py_GenericAlias(alias->origin, alias->args);
```

Comment:
This can return NULL, in which case the next line will crash.

Reply:
Oh, right, good point. I'm assuming the right thing to do if `starred_tuple` does end up being `NULL` is to just `return NULL` straight away?

Reply:
Yes you can just return NULL, Python will raise MemoryError for it.

Reply:
Ok, great.